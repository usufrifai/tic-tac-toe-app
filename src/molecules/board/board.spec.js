import React from 'react';
import Board from './index';
import {create, act} from 'react-test-renderer';
import Square from "../../atoms/square";

describe('Board', () => {
    it('Board should be rendered', () => {
        const component = create(
            <Board
            squares={['X', 'O', 'O', 'X', 'X', 'O', 'X', 'O', 'O']}
            onClick={jest.fn()}
            player= 'Rifay'
            />
          );
          let tree = component.toJSON();
          expect(tree).toMatchSnapshot();
    });

    it('Board should be rendered with Game Over message', () => {
        const component = create(
            <Board 
            winnerMessage="Game Draw"
            squares={['X', 'O', 'O', 'X', 'X', 'O', 'X', 'O', 'O']}
            onClick={jest.fn()}
            player= 'Rifay'
            />
          );
          const instance = component.root;
          const square = instance.findAllByType(Square);
          act(() => {
             square[0].props.onClick();

          });
          let tree = component.toJSON();
          expect(tree).toMatchSnapshot();
    });
  });