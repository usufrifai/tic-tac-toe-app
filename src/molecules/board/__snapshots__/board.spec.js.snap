// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board Board should be rendered 1`] = `
<div
  className="board-container"
>
  <div
    className="heading"
    style={
      Object {
        "width": undefined,
      }
    }
  >
    <h3>
      Game
    </h3>
  </div>
  <div
    className="board"
  >
    <div
      className="squares X"
      onClick={[Function]}
    >
      <span>
         
        X
         
      </span>
    </div>
    <div
      className="squares O"
      onClick={[Function]}
    >
      <span>
         
        O
         
      </span>
    </div>
    <div
      className="squares O"
      onClick={[Function]}
    >
      <span>
         
        O
         
      </span>
    </div>
    <div
      className="squares X"
      onClick={[Function]}
    >
      <span>
         
        X
         
      </span>
    </div>
    <div
      className="squares X"
      onClick={[Function]}
    >
      <span>
         
        X
         
      </span>
    </div>
    <div
      className="squares O"
      onClick={[Function]}
    >
      <span>
         
        O
         
      </span>
    </div>
    <div
      className="squares X"
      onClick={[Function]}
    >
      <span>
         
        X
         
      </span>
    </div>
    <div
      className="squares O"
      onClick={[Function]}
    >
      <span>
         
        O
         
      </span>
    </div>
    <div
      className="squares O"
      onClick={[Function]}
    >
      <span>
         
        O
         
      </span>
    </div>
  </div>
  <h5
    className="playerName"
  >
    Hey Rifay its your turn
  </h5>
</div>
`;

exports[`Board Board should be rendered with Game Over message 1`] = `
<div
  className="board-container"
>
  <div
    className="heading"
    style={
      Object {
        "width": undefined,
      }
    }
  >
    <h3>
      Game
    </h3>
  </div>
  <div
    className="board"
  >
    <div
      className="squares X"
      onClick={[Function]}
    >
      <span>
         
        X
         
      </span>
    </div>
    <div
      className="squares O"
      onClick={[Function]}
    >
      <span>
         
        O
         
      </span>
    </div>
    <div
      className="squares O"
      onClick={[Function]}
    >
      <span>
         
        O
         
      </span>
    </div>
    <div
      className="squares X"
      onClick={[Function]}
    >
      <span>
         
        X
         
      </span>
    </div>
    <div
      className="squares X"
      onClick={[Function]}
    >
      <span>
         
        X
         
      </span>
    </div>
    <div
      className="squares O"
      onClick={[Function]}
    >
      <span>
         
        O
         
      </span>
    </div>
    <div
      className="squares X"
      onClick={[Function]}
    >
      <span>
         
        X
         
      </span>
    </div>
    <div
      className="squares O"
      onClick={[Function]}
    >
      <span>
         
        O
         
      </span>
    </div>
    <div
      className="squares O"
      onClick={[Function]}
    >
      <span>
         
        O
         
      </span>
    </div>
  </div>
  <h5
    className="playerName"
  >
     Game Over 
  </h5>
</div>
`;
